### ISSUE TITLE
Unrecognized storage client

### ISSUE THREAD

**Issue by mrmodolo** [slim-bean/loki-shell#11]
Hi!

Today I updated loki-shell and now the service can not start!
All configuration files remain un-updated, only the binaries have been updated!

Below the output of the command **journalctl -xeu loki-shell.service**!

```
nov 19 09:33:54 nusakan loki[79949]: level=info ts=2020-11-19T12:33:54.069878218Z caller=main.go:128 msg="Starting Loki" version="(version=2.0.0, branch=HEAD, revision=6978ee5d)"
nov 19 09:33:54 nusakan loki[79949]: level=info ts=2020-11-19T12:33:54.072371406Z caller=server.go:225 http=0.0.0.0:4100 grpc=0.0.0.0:9095 msg="server listening on addresses"
nov 19 09:33:54 nusakan loki[79949]: level=info ts=2020-11-19T12:33:54.072825991Z caller=modules.go:420 msg="RulerStorage is not configured in single binary mode and will not be started."
nov 19 09:33:54 nusakan loki[79949]: level=error ts=2020-11-19T12:33:54.072873352Z caller=log.go:149 msg="error running loki" err="Unrecognized storage client , choose one of: aws, s3, gcs, azure, filesystem\nerror initialising module: compactor\ngithub.com/cortexproject/cortex/pkg/util/modules.(*Manager).initModule\n\t/src/loki/vendor/github.com/cortexproject/cortex/pkg/util/modules/modules.go:105\ngithub.com/cortexproject/cortex/pkg/util/modules.(*Manager).InitModuleServices\n\t/src/loki/vendor/github.com/cortexproject/cortex/pkg/util/modules/modules.go:75\ngithub.com/grafana/loki/pkg/loki.(*Loki).Run\n\t/src/loki/pkg/loki/loki.go:204\nmain.main\n\t/src/loki/cmd/loki/main.go:130\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1373"
nov 19 09:33:54 nusakan systemd[1]: loki-shell.service: Main process exited, code=exited, status=1/FAILURE

```

Thanks,
Marcelo Módolo

**Comment by mrmodolo** [comment]
Hi!

I found [Unrecognized storage client after schema upgrade](https://github.com/grafana/loki/issues/2815) and change my config to add this:

```
compactor:
  working_directory: /home/me/.cache/loki/boltdb-shipper-compactor
  shared_store: filesystem
```

My config file:

```
auth_enabled: false

server:
  http_listen_port: 4100

ingester:
  lifecycler:
    address: 127.0.0.1
    ring:
      kvstore:
        store: inmemory
      replication_factor: 1
    final_sleep: 0s
                        
  chunk_idle_period: 1h 
  max_chunk_age: 1h

  chunk_target_size: 1048576
  chunk_retain_period: 30s
  max_transfer_retries: 0

schema_config:
  configs:
    - from: 2020-08-08
      store: boltdb-shipper
      object_store: aws
      schema: v11
      index:
        prefix: index_
        period: 24h

compactor:
  working_directory: /home/me/.cache/loki/boltdb-shipper-compactor
  shared_store: filesystem

storage_config:
  aws:
    s3: https://XXXXXXXXXXXXXX:WIJDSJFJKOSDJKSDSDD@s3.amazonaws.com/MY-HISTORY-bucket 
    region: us-east-1 # FIXME Set the correct region
    s3forcepathstyle: true
  boltdb_shipper:
    active_index_directory: /home/me/.cache/loki/index 
    shared_store: aws
    cache_location: /home/me/.cache/loki/boltdb-cache 
    cache_ttl: 721h to improve performance.
limits_config:
  reject_old_samples: true
  reject_old_samples_max_age: 168h

chunk_store_config:
  max_look_back_period: 0s
  chunk_cache_config:
    enable_fifocache: true
    fifocache:
      max_size_bytes: 52428800

table_manager:
  retention_deletes_enabled: false
  retention_period: 0s

```
It is working again!

Thanks,
Marcelo Módolo

**Comment by slim-bean** [comment]
I really need to figure out how to make github notify my more aggressively, sorry I didn't see this sooner.

I'm super happy you found the issue and sorry I didn't see this sooner.

I just released Loki 2.1 today and there is a fix to explicitly log this with a better message.

Thanks for the issue and follow up, hope you are enjoying loki and loki-shell!
